miio/heater.py

```diff
@@ -0,0 +1,279 @@
+
+    def __init__(
+        self,
+        ip: str = None,
+        token: str = None,
+        start_id: int = 0,
+        debug: int = 0,
+        lazy_discover: bool = True,
+        model: str = MODEL_HEATER_ZA1,
+    ) -> None:
+        super().__init__(ip, token, start_id, debug, lazy_discover)
+
+        if model in AVAILABLE_PROPERTIES:
+            self.model = model
+        else:
+            self.model = MODEL_HEATER_ZA1
+
+    @command(
+        default_output=format_output(
+            "",
+            "Power: {result.power}\n"
+            "Target temperature: {result.target_temperature} °C\n"
+            "Temperature: {result.temperature} °C\n"
+            "Humidity: {result.humidity} %\n"
+            "Display brightness: {result.brightness}\n"
+            "Buzzer: {result.buzzer}\n"
+            "Child lock: {result.child_lock}\n"
+            "Power-off time: {result.delay_off_countdown}\n",
+        )
+    )
+    def status(self) -> HeaterStatus:
+        """Retrieve properties."""
+        properties = AVAILABLE_PROPERTIES[self.model]
+
+        # A single request is limited to 16 properties. Therefore the
+        # properties are divided into multiple requests
+        _props_per_request = 15
+
+        # The MA1, ZA1 is limited to a single property per request
+        if self.model in [MODEL_HEATER_MA1, MODEL_HEATER_ZA1]:
+            _props_per_request = 1
+
+        _props = properties.copy()
+        values = []
+        while _props:
+            values.extend(self.send("get_prop", _props[:_props_per_request]))
+            _props[:] = _props[_props_per_request:]
+
+        properties_count = len(properties)
+        values_count = len(values)
+        if properties_count != values_count:
+            _LOGGER.error(
+                "Count (%s) of requested properties does not match the "
+                "count (%s) of received values.",
+                properties_count,
+                values_count,
+            )
+
+        return HeaterStatus(dict(zip(properties, values)))
+
+    @command(default_output=format_output("Powering on"))
+    def on(self):
+        """Power on."""
+        return self.send("set_power", ["on"])
+
+    @command(default_output=format_output("Powering off"))
+    def off(self):
+        """Power off."""
+        return self.send("set_power", ["off"])
+
+    @command(
+        click.argument("temperature", type=int),
+        default_output=format_output("Setting target temperature to {temperature}"),
+    )
+    def set_target_temperature(self, temperature: int):
+        """Set target temperature."""
+        if self.model == MODEL_HEATER_ZA1 and (temperature < 16 or temperature > 32):
+            raise HeaterException("Invalid target temperature: %s" % temperature)
+        if self.model == MODEL_HEATER_MA1 and (temperature < 20 or temperature > 32):
+            raise HeaterException("Invalid target temperature: %s" % temperature)
```

Comment:
Would it be a good idea to have a dictionary storing these values (as well as those for the `delay_off`) to avoid branching? Maybe something like:
```
SUPPORTED_MODELS = {
    MODEL_HEATER_ZA1: {
        "max_temperature": 32,
        "min_temperature": 16,
        "max_delayoff": 9 * 3600,
    },
    MODEL_HEATER_MA1: { ... }
}
```

Reply:
done